### ISSUE TITLE
Add the ability to control the brush direction.

### ISSUE THREAD

**Issue by JeffOwOSun** [JeffOwOSun/Impressionist#9]
The stroke direction should be controlled **four** different ways: 
1. using a slider value, 
2. using the right mouse button to drag out a direction line, 
3. using the direction of the cursor movement, 
4. and using directions that are perpendicular to the gradient of the image. 

You can use a radio box to allow the user to select which method to use.

**Comment by JeffOwOSun** [comment]
Oh, I discovered that in order to add "set angle by right mouse click" function, I need to remember the LastPoint right within the Draw function, RIGHT_BUTTON_DOWN section. Because that's not appropriate to be put into any existing brush, nor is it appropriate to be put into a new brush, I decided to store it as a private attribute of the PaintView Class. @Oh233 You may consider directly use this one for your gradient-determined brush direction

**Comment by HaozhiQi** [comment]
Is what you want just the _current_ location of current mouse? Maybe you can call Fl::event_x() and Fl::event_y() instead of setting another attribute? That is stored globally I think. 

If not, then nevermind

**Comment by JeffOwOSun** [comment]
Okay in the newest commit 8c3b2e164881f65598f257d95e07f9e6d63ad35e I resolved point 2 mentioned above.

**Note** that I created a m_ptLastPoint private attribute in paintView class. You may wish to log the last point there in your implementation of point 3.

**Then** I'll be looking into point 4. I'll divide it into the following steps:
1. Create a function to apply the gaussian filter to the image at a particular spot
2. Create a function to apply the 1 -1 and 1, -1 matrix to extract the gradient at a particular spot
3. Then in line brush as well as in scattered line brush, when ever a line is drawn, we do 1, 2 to the source pixel (probably its neighborhood as well) and then calculate the angle for the line

**Now the problem is** we are not going to apply the filters and update the framebuffer itself. We just need a _small patch_ of the image filtered, and we require the paintView framebuffer is _not_ altered. Therefore we should come up with a way to store the patch of image.

**Comment by JeffOwOSun** [comment]
@Oh233 **Pay attention!** I saw your attempt to directly look up `Fl::event_x()` and `Fl::event_y()`, but pay attention, the coordinating system is a little bit different from the rendering coordinate. You can check it up in `PaintView.cpp`, line 103 and 104 for a conversion from Fl::event_y() to target and source y.

Here is the code:

``` C++
//coord.x and y come directly from Fl::event_x() and y
Point source( coord.x + m_nStartCol, m_nEndRow - coord.y ); // Inverting coord.y
Point target( coord.x, m_nWindowHeight - coord.y ); // Inverting coord.y
```

**Comment by JeffOwOSun** [comment]
Solved point 4 in f0338888656c0ff8814ccbc629dd7447d6c0d768

**Comment by JeffOwOSun** [comment]
**Still a problem**, the gradient is only made for LineBrush. We need to extend it to Scattered Line
